dotnet/src/SemanticKernel.Skills/Skills.OpenAPI/Model/RestApiOperation.cs

```diff
@@ -101,6 +104,61 @@ public Uri BuildOperationUrl(IDictionary<string, string> arguments)
         return new Uri($"{serverUrl.TrimEnd('/')}/{path.TrimStart('/')}", UriKind.Absolute);
     }
 
+    /// <summary>
+    /// Renders operation request headers.
+    /// </summary>
+    /// <param name="arguments">The operation arguments.</param>
+    /// <returns>The rendered request headers.</returns>
+    public IDictionary<string, string> RenderHeaders(IDictionary<string, string> arguments)
+    {
+        var headers = new Dictionary<string, string>();
+
+        foreach (var header in this.Headers)
+        {
+            var headerName = header.Key;
+            var headerValue = header.Value;
+
+            //A try to resolve header value in arguments.
+            if (arguments.TryGetValue(headerName, out var value))
+            {
+                headers.Add(headerName, value);
+                continue;
+            }
+
+            //Header value is already supplied.
+            if (!string.IsNullOrEmpty(headerValue))
+            {
+                headers.Add(headerName, headerValue);
+                continue;
+            }
+
+            //Getting metadata for the header
+            var headerMetadata = this.Parameters.FirstOrDefault(p => p.Location == RestApiOperationParameterLocation.Header && p.Name == headerName);
+            if (headerMetadata == null)
+            {
+                //No metadata found for the header.
+                throw new RestApiOperationException($"No value for the '{headerName} header is found.'");
+            }
+
+            //If parameter is required it's value should always be provided.
+            if (headerMetadata.IsRequired)
```

Comment:
Don't have strong preferences here, even so more inclined to the current approach because of the clarity - reading the code  
 from the top to the bottom, block by block, without parsing || conditions in if statements is very easy :)

Reply:
Makes sense, thanks!